src/server/scripts/Zandalar/Underrot/boss_sporecaller_zancha.cpp

```diff
@@ -0,0 +1,707 @@
+    SPELL_TELEPORT_BEHIND_YOU_SELECTOR = 272786,
+    SPELL_TELEPORT                     = 272787,
+    SPELL_THOUSAND_FOLD_BLADE          = 272864
+};
+
+enum SporecallerZanchaEvents
+{
+    EVENT_CHECK_ENERGY = 1,
+    EVENT_SHOCKWAVE,
+    EVENT_UPHEAVAL,
+    EVENT_VOLATILE_PODS,
+    EVENT_BOUNDLESS_ROT,
+
+    // Musashitake
+    EVENT_TELEPORT
+};
+
+enum SporecallerZanchaTexts
+{
+    SAY_AGGRO             = 0,
+    SAY_UPHEAVAL          = 1,
+    SAY_FESTERING_HARVEST = 2,
+    SAY_WIPE              = 3,
+    SAY_KILL              = 4,
+    SAY_DEATH             = 5,
+
+    // Musashitake
+    SAY_PERSONAL          = 0
+};
+
+enum SporecallerZanchaSummonGroups
+{
+    SUMMON_GROUP_ZANCHA_INTRO  = 0,
+    SUMMON_GROUP_MUSASHITAKE   = 1,
+    SUMMON_GROUP_VOLATILE_PODS = 2
+};
+
+enum SporecallerZanchaActions
+{
+    ACTION_INTRO = 1
+};
+
+enum SporecallerZanchaCreatures
+{
+    NPC_SPORE_POD   = 131597
+};
+
+enum MusashitakeMisc
+{
+    CONVERSATION_START = 8358
+};
+
+constexpr Position CenterPosition = { 1032.9444f, 1058.2899f, 33.330894f };
+
+// 131383 - Sporecaller Zancha
+struct boss_sporecaller_zancha : public BossAI
+{
+    boss_sporecaller_zancha(Creature* creature) : BossAI(creature, DATA_SPORECALLER_ZANCHA), _shockwaveCount(1), _upheavalCount(1), _volatilePodsCount(1) { }
+
+    void JustAppeared() override
+    {
+        me->SummonCreatureGroup(SUMMON_GROUP_MUSASHITAKE);
+        me->SetPowerType(POWER_ENERGY);
+        me->SetPower(POWER_ENERGY, 0);
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+        if (!victim->IsPlayer())
+            return;
+
+        Talk(SAY_KILL);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+
+        Talk(SAY_DEATH);
+    }
+
+    void Reset() override
+    {
+        _Reset();
+
+        _shockwaveCount = 1;
+        _upheavalCount = 1;
+        _volatilePodsCount = 1;
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        summons.DespawnAll();
+
+        Talk(SAY_WIPE);
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me, 1);
+
+        Talk(SAY_AGGRO);
+
+        DoCastSelf(SPELL_BOUNDLESS_ROT);
+        DoCastSelf(SPELL_FESTERING_HARVEST);
+        events.ScheduleEvent(EVENT_CHECK_ENERGY, 500ms);
+
+        if (IsHeroicOrHigher() || IsMythicPlus())
+        {
+            events.ScheduleEvent(EVENT_SHOCKWAVE, 10600ms);
+            events.ScheduleEvent(EVENT_UPHEAVAL, 16600ms);
+            events.ScheduleEvent(EVENT_VOLATILE_PODS, 21600ms);
+        }
+        else
+        {
+            events.ScheduleEvent(EVENT_SHOCKWAVE, 10500ms);
+            events.ScheduleEvent(EVENT_UPHEAVAL, 16s);
```

Comment:
if we're going for the ms only, then 16000ms or 16s and then we change the rest to 10s + 500ms, and such.